### PR TITLE
feat: start sync automagically (AR-1924)

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We need to call `SyncManager` everywhere in the code.

### Solutions

- Start using the newly added mechanisms, like `SyncCriteriaProvider` to start Sync automatically.
- Disable `startSyncIfIdle` and deprecate similar functions in `SyncManager` so they can be removed in a future PR.

#### Test Coverage

- [X] I have added automated test to this contribution

Also, tested manually on Reloaded using these changes.

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SyncManager.kt [hunk 1]
```diff
@@ -30,27 +29,55 @@ interface SyncManager {
      * @see startSyncIfIdle
      * @see waitUntilSlowSyncCompletion
      */
+    @Deprecated(
+        message = "SyncManager won't serve as a Sync Utils anymore",
+        ReplaceWith(
+            expression = "syncRepository.syncState.first { it is SyncState.Live }",
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SyncManager.kt:35]: Seems like we will need to do this in quite few places. Would it make sense to move the `waitUntilLive` to the SyncRepository?

Comment (vitorhugods) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SyncManager.kt:35]: Indeed.
Maybe I was too aggressive with the changes.
I've noticed this in #720 and ended up not addressing the `waitUntilLive`.

Maybe I can just remove the deprecation for this one for now? What do you think?
To be honest I don't mind this being in `SyncManager`. Maybe later we can move somewhere else if it makes sense.

Comment (typfel) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/SyncManager.kt]: I agree, let's remove the deprecation.

**Review by typfel — APPROVED** [review]

**Review by alexandreferris — APPROVED** [review]
👏🏻

**Pull request merged**